librz/util/table.c

```diff
@@ -195,53 +195,84 @@ RZ_API void rz_table_set_columnsf(RzTable *t, const char *fmt, ...) {
 	va_end(ap);
 }
 
+#define add_column_to_rowf(row, fmt, ap) do { \
+		const char *arg = NULL; \
+		switch (fmt) { \
+		case 's': \
+		case 'z': \
+			arg = va_arg(ap, const char *); \
+			rz_list_append(row, strdup(arg ? arg : "")); \
+			break; \
+		case 'b': \
+			rz_list_append(row, rz_str_new(rz_str_bool(va_arg(ap, int)))); \
+			break; \
+		case 'i': \
+		case 'd': \
+			rz_list_append(row, rz_str_newf("%d", va_arg(ap, int))); \
+			break; \
+		case 'n': \
+			rz_list_append(row, rz_str_newf("%" PFMT64d, va_arg(ap, ut64))); \
+			break; \
+		case 'u': \
+			rz_list_append(row, rz_num_units(NULL, 32, va_arg(ap, ut64))); \
+			break; \
+		case 'x': \
+		case 'X': { \
+			ut64 n = va_arg(ap, ut64); \
+			if (n == UT64_MAX) { \
+				if (fmt == 'X') { \
+					rz_list_append(row, strdup("----------")); \
+				} else { \
+					rz_list_append(row, strdup("-1")); \
+				} \
+			} else { \
+				if (fmt == 'X') { \
+					rz_list_append(row, rz_str_newf("0x%08" PFMT64x, n)); \
+				} else { \
+					rz_list_append(row, rz_str_newf("0x%" PFMT64x, n)); \
+				} \
+			} \
+		} break; \
+		default: \
+			eprintf("Invalid format string char '%c', use 's' or 'n'\n", fmt); \
+			break; \
+		} \
+	} while(0)
+
+/**
+ * Add some columns values to the last created row, if any, or create a new row otherwise.
+ */
+RZ_API void rz_table_add_row_columnsf(RzTable *t, const char *fmt, ...) {
```

Comment:
What if `fmt` is NULL? Please add corresponding check

Reply:
fmt must be checked. we will have warnings on this when compiles

Reply:
Do you mean `RZ_PRINTF_CHECK()` macro? It will not work since the table `fmt` is different from `printf()`-like formats.